### ISSUE TITLE
Add ability to change Progress Bar background color.

### ISSUE THREAD

**Issue by demongod77** [adafruit/Adafruit_CircuitPython_ProgressBar#20]
Currently the background color is hardcoded in the library.

**Comment by FoamyGuy** [comment]
you can change the colors used with parameters in the constructor. See the magtag example script here for an example of using custom colors:
https://github.com/adafruit/Adafruit_CircuitPython_ProgressBar/blob/f8206d40e9375bfa5ffc8ace2948751c742c8f8f/examples/progressbar_magtag_simpletest.py#L35

The full init is:
```python
progress_bar = ProgressBar(
    x, y, BAR_WIDTH, BAR_HEIGHT, 1.0, bar_color=0x666666, outline_color=0xFFFFFF
)
```

**Comment by demongod77** [comment]
When I say background color, I am talking about the rightmost portion when the progress is less than 100%.

**Comment by FoamyGuy** [comment]
Ah I see what you mean, I do think it would be good to allow it to be set to any color instead of hard-coded to `0x0`. 

Are you interested in making a PR with this change? We can help you along the way if you are new to git or github. 

If not I can try to work on this enhancement later this week or next.

**Comment by demongod77** [comment]
Yes, I would like to do the pull request, but would need some help walking me through how to so.

**Comment by FoamyGuy** [comment]
Awesome! Thank you for being willing to contribute. For now I will link to you some great guides, and ask a few more questions so I can try to get a good understanding of your current level of experience with it so I know how to best help.

Taking a quick step back... Do you have an idea of what needs to change within the library code in order to support custom backgrounds? or would you like a point in the right direction on that front as well? 

These guides cover git / github and the other tools that are needed when working on Circuit Python related libraries:

- Comprehensive guide to contributing to Circuit Python with git and github: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github This goes through step by step creating the PR and all steps needed.

- Sharing a library guide: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library this one contains more information about some of the secondary tools and utilities used by the repositories. PyLint and Black are the main two that you'll need to worry about for this I think.

- lastly for now the Circuit Python discord: https://adafru.it/discord there is a `#help-with-circuitpython` channel in that discord where myself and many other folks hang out and can offer up help if you get stuck on anything along the way.

**Comment by demongod77** [comment]
I have made the update to my local copy and seems to be working correctly.  I will read through those guides over the next few days.

**Comment by demongod77** [comment]
Other things came up, I am not going to be able to do my own pull request on this for a while.  I can provide my version with the update I did locally.

**Comment by FoamyGuy** [comment]
No worries. That would be great if you can provide your version. 

I'll work on getting this implemented.

**Comment by demongod77** [comment]
What would be the best way to do so?  Copy here or attach?

**Comment by FoamyGuy** [comment]
whatever is easiest for you honestly. you can paste here if you put triple back ticks around it (shift tilda key) it will get formatted:
```
# example
```
or you can try attaching the file. Maybe make it a txt file first, there may be restriction on type.

**Comment by demongod77** [comment]
Lets see how this goes.
[adafruit_progressbar.txt](https://github.com/adafruit/Adafruit_CircuitPython_ProgressBar/files/5898975/adafruit_progressbar.txt)

**Comment by FoamyGuy** [comment]
Perfect! Thank You for that! I'll work on getting a PR in with this.

**Comment by demongod77** [comment]
Much thanks.

**Comment by hugodahl** [comment]
This is one of the changes I've made as part of issue #21. While I intended to address it as its own separate issue, I got into it as I was refactoring and "in the plumbing".

**Comment by hugodahl** [comment]
This has been added as of v2.0 of the ProgressBar project when using either the `HorizontalProgressBar` or `VerticalProgressBar` widgets, and setting the `fill_color` property.

**Comment by hugodahl** [comment]
I recommend we close this issue as resolved, @CircuitPythonLibrarians.

**Comment by jposada202020** [comment]
Closed by #21
